addons/web_editor/models/ir_attachment.py

```diff
@@ -84,3 +84,16 @@ def _can_bypass_rights_on_media_dialog(self, **attachment_data):
         - Non admin user uploading an unsplash image (bypass binary/url check)
         """
         return False
+
+    def _is_used_in_html_field(self):
+        """Returns a model where this attachment is used, or a dict with model names or False."""
+        url = tools.html_escape(self.local_url)
+        views = self.env['ir.ui.view'].search([
+            "|",
+            ('arch_db', 'like', f'"{url}"'),
+            ('arch_db', 'like', f"'{url}'")
+        ])
+        return {
+            'matches': views,
+            'sudo_models': [],
+        } if views else None
```

Comment:
So this is a "light" version of what website is doing with the new table, so you have a limited way of knowing if the attachment is used?

But since the web_editor is now used in a lot of modules in Odoo, searching only in `ir.ui.view` seems close to useless IMO.
What about tasks, knowledge, mass mailing (surely less important for this one) and so on?

What I would so IMHO is:
- Either move this deletion check feature in website
- Either add this new table in `web_editor` so it benefits from this full check?

Reply:
Turned in into a `web_editor` feature.